appservice/src/tree/DeploymentTreeItem.ts

```diff
@@ -55,14 +56,36 @@ export class DeploymentTreeItem extends AzureTreeItem<ISiteTreeRoot> {
         return this._deployResult.id;
     }
 
+    public get description(): string | undefined {
+        if (this._deployResult.active) {
+            return localize('active', 'Active');
+        }
+
+        switch (this._deployResult.status) {
+            case DeployStatus.Building:
+                return localize('building', 'Building...');
+            case DeployStatus.Deploying:
+                return localize('deploying', 'Deploying...');
+            case DeployStatus.Pending:
+                return localize('pending', 'Pending...');
+            case DeployStatus.Failed:
+                return localize('failed', 'Failed');
+            case DeployStatus.Success:
+            default:
+                return;
+        }
+    }
+
     public async redeployDeployment(): Promise<void> {
         const redeploying: string = `Redeploying commit "${this.id}" to "${this.root.client.fullName}"`;
         const deployed: string = `Commit "${this.id}" has been redeployed to "${this.root.client.fullName}".`;
         window.withProgress({ location: ProgressLocation.Notification, title: redeploying }, async (): Promise<void> => {
             const kuduClient: KuduClient = await getKuduClient(this.root.client);
             ext.outputChannel.appendLine(redeploying);
+            const refreshingInteveral: NodeJS.Timer = setInterval(async () => { await this.refresh(); }, 1000); /* the status of the label changes during deployment so poll for that*/
```

Comment:
Wrap this in a try/finally so that `clearInterval` is called even if `await kuduClient.deployment.deploy(this.id);` throws an exception